using_restricted_array.rb

```diff
@@ -1,47 +1,93 @@
 require_relative 'restricted_array.rb'
-# RestrictedArray can be created using a specified size, or a random size in
-# the range of 1-20 will be chosen for you.
+# RestrictedArray can be created using a specified size, or a random size in the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
 def length(array)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  until array == nil
```

Comment:
This should be `until array[i] == nil` I'm surprised that the current code doesn't result in an infinite loop.